docs/user-guide.md

```diff
@@ -0,0 +1,856 @@
+packages = ["arrr", "numberwang", "snowballstemmer>=2.2.0" ]
+```
+
+```JSON title="A packages list in JSON"
+{
+    "packages": ["arrr", "numberwang", "snowballstemmer>=2.2.0" ]
+}
+```
+
+!!! warning
+
+    Because `micropip` is a Pyodide-only feature, and MicroPython doesn't
+    support code packaged on PyPI, **the `packages` option is only available
+    if you use Pyodide as your interpreter**.
+
+The names in the list of `packages` can be any of the following valid forms:
+
+* A name of a package on PyPI: `"snowballstemmer"`
+* A name for a package on PyPI with additional constraints: `"snowballstemmer>=2.2.0"`
+* An arbitrary URL to a Python package: `"https://.../package.whl"`
+* A file copied onto the browser based file system: `"emfs://.../package.whl"`
+
+#### Plugins
+
+The `plugins` enumerate plugins enabled by PyScript to add extra functionality
+to the platform.
+
+Each plugin should be included on the web page, as described in the
+[plugins](#plugins) section below. Then the plugin's name should be listed.
+
+```TOML title="A list of plugins in TOML"
+plugins = ["custom_plugin", "!error"]
+```
+
+```JSON title="A list of plugins in JSON"
+{
+    "plugins": ["custom_plugin", "!error"]
+}
+```
+
+!!! info
+
+    The `"!error"` syntax is a way to turn off a plugin built into PyScript
+    that is enabled by default.
+
+    Currently, the only built-in plugin is the `error` plugin (which displays a
+    stack trace and error messages in the DOM). More may be added at a later
+    date.
+
+#### Custom 
+
+Sometimes plugins or apps need bespoke configuration options.
+
+So long as you don't cause a name collision with the built-in option names then
+you are free to use any valid data structure that works with both TOML and JSON
+to express your configuration needs.
+
+**TODO: explain how to programmatically get access to an object representing
+the config.**
+
+## The DOM
+
+The DOM
+([document object model](https://developer.mozilla.org/en-US/docs/Web/API/Document_object_model))
+refers to a tree like data structure that represents the web page in the
+browser. PyScript needs to be able to interact with the DOM in order to change
+the user interface and react to things happening in the browser.
+
+There are currently two ways to interact with the DOM:
+
+1. Through the foreign function interface (FFI) and by directly interacting
+   with the objects found in the browser's `globalThis` object.
+2. Through the `pydom` module that comes as standard with PyScript.
+
+### FFI
+
+The foreign function interface (FFI) gives Python access to all the
+[standard web capabilities and features](https://developer.mozilla.org/en-US/docs/Web),
+such as the browser's built-in
+[web APIs](https://developer.mozilla.org/en-US/docs/Web/API).
+
+This is available via the `pyscript.window` module which is a proxy for
+JavaScript's `globalThis` object, or `pyscript.document` which is a proxy for
+the `document` object in JavaScript:
+
```

Comment:
this is not true in workers.
`import js` is always a `globalThis` in both cases.
In workers, `pyscript.window` is a proxy to the main thread window

Reply:
Good catch. Let me change this.